airbyte-server/src/main/java/io/airbyte/server/handlers/OAuthHandler.java

```diff
@@ -321,23 +321,25 @@ Map<String, String> buildJsonPathFromOAuthFlowInitParameters(final Map<String, L
 
   @VisibleForTesting
   JsonNode getOauthFromDBIfNeeded(final JsonNode oAuthInputConfigurationFromDB, final JsonNode oAuthInputConfigurationFromInput) {
-    final Map<String, String> result = new HashMap<>();
-
-    Jsons.deserializeToStringMap(oAuthInputConfigurationFromInput)
-        .forEach((k, v) -> {
-          if (AirbyteSecretConstants.SECRETS_MASK.equals(v)) {
-            if (oAuthInputConfigurationFromDB.has(k)) {
-              result.put(k, oAuthInputConfigurationFromDB.get(k).textValue());
-            } else {
-              LOGGER.warn("Missing the key {} in the config store in DB", k);
-            }
-
-          } else {
-            result.put(k, v);
-          }
-        });
-
-    return Jsons.jsonNode(result);
+    final ObjectNode result = (ObjectNode) Jsons.emptyObject();
+
+    oAuthInputConfigurationFromInput.fields().forEachRemaining(entry -> {
+      final String k = entry.getKey();
+      final JsonNode v = entry.getValue();
+
+      // Note: This does not currently handle replacing masked secrets within nested objects.
```

Comment:
I noticed this, but it's not an existing issue. I verified with the previous behavior and if there was a nested object in the input configuration, `deserializeToStringMap` actually threw an error.